rosbag2/src/rosbag2/writer.cpp

```diff
@@ -132,6 +146,28 @@ void Writer::remove_topic(const TopicMetadata & topic_with_type)
   }
 }
 
+void Writer::split_bagfile()
+{
+  const auto storage_uri = format_storage_uri(
+    base_folder_,
+    metadata_.relative_file_paths.size());
```

Comment:
One issue I came across with the decompress is that I had to add 1 to the size since the file wasn't pushed into the relative_file_paths prior to `format_storage_uri` being called. Not sure if that's addressed here.

Reply:
Are you basing off of an older version? metadata's `relative_file_paths` is initialized with the initial storage's `relative_file_path` which is formatted with `storage_count=0`.

The unit test I wrote should be validating that the relative file paths are incrementing from 0 to n